src/main/java/io/vertx/redis/impl/RedisConnection.java

```diff
@@ -248,10 +248,14 @@ void send(final Command<?> command) {
         break;
       case CONNECTING:
       case ERROR:
-        pending.add(command);
+        runOnContext(v -> {
+          pending.add(command);
```

Comment:
it should actually be a real synchronization without `runOnContext` because you can run in the case with `send` can be called two times: the `pending` list is processed while having context tasks that will add command to the pending queue.

and this should be done along with the `state` field, i.e you append to `pending` only if the state is CONNECTING or ERROR and at connection time, the state is updated to CONNECTED and the pending queue is emptied atomically (which means we should not use an AtomicReference.